doc/RE

```diff
@@ -375,7 +375,7 @@ syntax: ONIG_SYNTAX_RUBY (default)
     \k<name+level> \k'name+level'
     \k<name-level> \k'name-level'
 
-    Destine a group on the recursion level relative to the referring position.
+    Define a group on the recursion level relative to the referring position.
```

Comment:
I don't know the intention of the original author, but `\k<>` doesn't define a group. It's a back reference, so it's refer a group.
How about changing `Destine` to `Refer`?